Version of openjp2 causing issue in new projects

This is an issue I am only getting on new projects. I have the same dependencies including version in an older project that works without error. Below is a simplified set of steps that can reproduce the issue. I did notice different versions of sub-dependencies in Cargo.lock file. Specifically in older project, with the same dependency versions I get:

jpeg2k: 0.6.6 vs 0.6.7
openjpy: 0.3.0 vs 0.3.4

To replicate (running on Linux, cargo 1.76.0 ):
 - cargo new test-project
 - cd test-project
 - cargo add --features pixeldata
 

This creates the following dependencies in the Cargo.toml
[dependencies]
dicom = { version = "0.6.3", features = ["pixeldata"] }


Then run:
 - cargo check

    Checking openjp2 v0.3.4
error[E0425]: cannot find value `opj_stream_read_skip` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:664:7
    |
664 |       opj_stream_read_skip
    |       ^^^^^^^^^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_read_skip;
    |

error[E0425]: cannot find value `opj_stream_read_seek` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:672:7
    |
672 |       opj_stream_read_seek
    |       ^^^^^^^^^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_read_seek;
    |

error[E0425]: cannot find value `opj_stream_write_skip` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:682:7
    |
682 |       opj_stream_write_skip
    |       ^^^^^^^^^^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_write_skip;
    |

error[E0425]: cannot find value `opj_stream_write_seek` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:690:7
    |
690 |       opj_stream_write_seek
    |       ^^^^^^^^^^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_write_seek;
    |

error[E0425]: cannot find value `opj_stream_default_read` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:699:5
    |
699 |     opj_stream_default_read
    |     ^^^^^^^^^^^^^^^^^^^^^^^
...
726 | pub unsafe extern "C" fn opj_stream_default_create(mut l_is_input: OPJ_BOOL) -> *mut opj_stream_t {
    | ------------------------------------------------------------------------------------------------- similarly named function `opj_stream_default_create` defined here
    |
help: a function with a similar name exists
    |
699 |     opj_stream_default_create
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_default_read;
    |

error[E0425]: cannot find value `opj_stream_default_write` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:707:5
    |
707 |     opj_stream_default_write
    |     ^^^^^^^^^^^^^^^^^^^^^^^^
...
726 | pub unsafe extern "C" fn opj_stream_default_create(mut l_is_input: OPJ_BOOL) -> *mut opj_stream_t {
    | ------------------------------------------------------------------------------------------------- similarly named function `opj_stream_default_create` defined here
    |
help: a function with a similar name exists
    |
707 |     opj_stream_default_create
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_default_write;
    |

error[E0425]: cannot find value `opj_stream_default_skip` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:715:5
    |
715 |     opj_stream_default_skip
    |     ^^^^^^^^^^^^^^^^^^^^^^^
...
726 | pub unsafe extern "C" fn opj_stream_default_create(mut l_is_input: OPJ_BOOL) -> *mut opj_stream_t {
    | ------------------------------------------------------------------------------------------------- similarly named function `opj_stream_default_create` defined here
    |
help: a function with a similar name exists
    |
715 |     opj_stream_default_create
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_default_skip;
    |

error[E0425]: cannot find value `opj_stream_default_seek` in this scope
   --> /home/chris/.cargo/registry/src/index.crates.io-6f17d22bba15001f/openjp2-0.3.4/src/openjpeg.rs:719:5
    |
719 |     opj_stream_default_seek
    |     ^^^^^^^^^^^^^^^^^^^^^^^
...
726 | pub unsafe extern "C" fn opj_stream_default_create(mut l_is_input: OPJ_BOOL) -> *mut opj_stream_t {
    | ------------------------------------------------------------------------------------------------- similarly named function `opj_stream_default_create` defined here
    |
help: a function with a similar name exists
    |
719 |     opj_stream_default_create
    |
help: consider importing this function
    |
34  + use crate::cio::opj_stream_default_seek;
    |

For more information about this error, try `rustc --explain E0425`.
error: could not compile `openjp2` (lib) due to 8 previous errors

Please let me know if I you would like me to supply any more information on this issue? Any help on this would be greatly appreciated.

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -25,6 +25,8 @@ jobs:
           cache: true
       # test project with default + extra features
       - run: cargo test --features image,ndarray,sop-class,rle,cli
+      # test dicom-pixeldata with openjp2
+      - run: cargo test -p dicom-pixeldata --features openjp2
       # test dicom-pixeldata with openjpeg-sys
       - run: cargo test -p dicom-pixeldata --features openjpeg-sys
       # test dicom-pixeldata with gdcm-rs

--- a/transfer-syntax-registry/Cargo.toml
+++ b/transfer-syntax-registry/Cargo.toml
@@ -23,7 +23,7 @@ native_windows = ["jpeg", "rle"]
 jpeg = ["jpeg-decoder", "jpeg-encoder"]
 # JPEG 2000 support via the OpenJPEG Rust port,
 # works on Linux and a few other platforms
-openjp2 = ["dep:jpeg2k", "jpeg2k/openjp2"]
+openjp2 = ["dep:jpeg2k", "dep:openjp2", "jpeg2k/openjp2"]
 # native RLE lossless support
 rle = []
 # enable Rayon for JPEG decoding

--- a/transfer-syntax-registry/Cargo.toml
+++ b/transfer-syntax-registry/Cargo.toml
@@ -31,7 +31,7 @@ rayon = ["jpeg-decoder?/rayon"]
 # enable SIMD operations for JPEG encoding
 simd = ["jpeg-encoder?/simd"]
 
-# JPEG 2000 support via the OpenJPEG Rust port,
+# JPEG 2000 support via the OpenJPEG native bindings,
 # conflicts with `openjp2`
 openjpeg-sys = ["dep:jpeg2k", "jpeg2k/openjpeg-sys"]
 
@@ -48,6 +48,11 @@ tracing = "0.1.34"
 
 [dependencies.jpeg2k]
 version = "0.6.6"
+optional = true
+default-features = false
+
+[dependencies.openjp2]
+version = "0.3.5"
 optional = true
 default-features = false
 

--- a/transfer-syntax-registry/Cargo.toml
+++ b/transfer-syntax-registry/Cargo.toml
@@ -23,7 +23,7 @@ native_windows = ["jpeg", "rle"]
 jpeg = ["jpeg-decoder", "jpeg-encoder"]
 # JPEG 2000 support via the OpenJPEG Rust port,
 # works on Linux and a few other platforms
-openjp2 = ["dep:jpeg2k", "dep:openjp2", "jpeg2k/openjp2"]
+openjp2 = ["dep:jpeg2k", "jpeg2k/openjp2"]
 # native RLE lossless support
 rle = []
 # enable Rayon for JPEG decoding
@@ -47,12 +47,7 @@ byteordered = "0.6"
 tracing = "0.1.34"
 
 [dependencies.jpeg2k]
-version = "0.6.6"
-optional = true
-default-features = false
-
-[dependencies.openjp2]
-version = "0.3.6"
+version = "0.6.8"
 optional = true
 default-features = false
 

--- a/transfer-syntax-registry/Cargo.toml
+++ b/transfer-syntax-registry/Cargo.toml
@@ -52,7 +52,7 @@ optional = true
 default-features = false
 
 [dependencies.openjp2]
-version = "0.3.5"
+version = "0.3.6"
 optional = true
 default-features = false
 

